app/controllers/plants_controller.rb

```diff
@@ -0,0 +1,71 @@
+class PlantsController < ApplicationController
+  before_action :set_plant, only: [:show, :edit, :update, :destroy]
+
+  def show
+  end
+
+  def new
+    @plant = Plant.new
+    
+  end
+
+  def index
+    @plants = Plant.all
+  end
+
+  # GET /plants/1/edit
+  def edit
+  end
+
+  # POST /plants
+  # POST /plants.json
+  def create
+    @plant = Plant.new(plant_params)
+
+
+    respond_to do |format|
+      if @plant.save
+        format.html { redirect_to @plant, notice: 'Roślinkę dodano poprawnie.' }
+        format.json { render :show, status: :created, location: @plant }
+      else
+        format.html { render :new }
+        format.json { render json: @plant.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /plant/1
+  # PATCH/PUT /plants/1.json
+  def update
+    respond_to do |format|
+      if @plant.update(plant_params)
+        format.html { redirect_to @plant, notice: 'Roślinka została zapdejtowana.' }
+        format.json { render :show, status: :ok, location: @plant }
+      else
+        format.html { render :edit }
+        format.json { render json: @plant.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /plants/1
+  # DELETE /plants/1.json
+  def destroy
+    @plant.destroy
+    respond_to do |format|
+      format.html { redirect_to plants_url, notice: 'Skasowałeś roślinkę :).' }
+      format.json { head :no_content }
+    end
+  end
+
+private
```

Comment:
Please correct indentation and add a blank line after this one :blush:

Reply:
Please fix indentation :joy:
```suggestion
  private
```